### ISSUE TITLE
Meteor.pdf.save() returns null when deployed

### ISSUE THREAD

**Issue by rico** [jgdovin/meteor-html-pdf#10]
I'm not really sure where to ask - but maybe someone here has an idea.

I'm using `Meteor.pdf.save()` on my local dev host without any problem. However, when the application is deployed to the Server using [Meteor Up](https://github.com/arunoda/meteor-up) the Method returns a file with just `null` in it.

Any Idea why this happens? I'm using node v0.10.36.

**Comment by silverfangs** [comment]
Rico, I have encountered the similar situation, do u have any idea about this?

**Comment by rico** [comment]
Unfortunately not. Maybe if I figure out how to run the application on the server without using Meteor up, I can spot an error.

**Comment by rico** [comment]
I read in a few comments that switching from mup to [mupx](https://github.com/arunoda/meteor-up/tree/mupx) solved different problems. I try to switch my deployment as well, but still running into a few problems. However, [this](http://code.krister.ee/mupx-problems/) helped a lot already.

**Comment by rico** [comment]
To rule out problems with mup / mupx, I manually deployed the application to a new server using the information found [here](https://www.digitalocean.com/community/tutorials/how-to-deploy-a-meteor-js-application-on-ubuntu-14-04-with-nginx).

However, the problem is still the same. So I guess it must have something to do how this package behaves when the application is built for production.

**Comment by rico** [comment]
Dear @jgdovin  - any information or help would be very much appreciated.

@silverfangs - did you find any workaround, alternative package or whatever?

**Comment by silverfangs** [comment]
Dear @rico , yes I do, I find that html-pdf rendering from [ongoworks:pdf](https://github.com/jgdovin/meteor-html-pdf) is very handy.
The end-product is the same, just that in ongoworks:pdf, you placed your desired output html in meteor template form.

The only problem (or maybe a good thing) is that, you have to configure the position and size of the pdf to fit to your desired pdf view.

**Comment by rico** [comment]
Thanks @silverfangs. However, the styling seems not to be as simple as with this package.

**Comment by silverfangs** [comment]
@rico , yes, a lot of work are required for the styling.

**Comment by srokatonie** [comment]
@rico @silverfangs  any luck with that?

**Comment by rico** [comment]
@srokatonie no, unfortunately not. As a _workaround_, I use a PHP service (API) which creates the PDF.

**Comment by srokatonie** [comment]
I came up with something that might be useful. It's based on [https://forums.meteor.com/t/send-pdf-from-server-to-client/6246/17](https://forums.meteor.com/t/send-pdf-from-server-to-client/6246/17)
Basically generate pdf in a public folder and then open it in client (I use Angular-Meteor but you can get the general idea).

Method:

```
import { Meteor } from 'meteor/meteor';
import htmlPdf from 'html-pdf';
import Future from 'fibers/future';

export function foo(html) {
  let fut = new Future();
  let filename;

  if (Meteor.isProduction) {
    // hardcoded to come up with some other way of getting path
    filename = '/var/www/myapp/bundle/programs/web.browser/app/tmp/Report.pdf';
  } else {
    filename = Meteor.absolutePath + '/public/tmp/Report.pdf';
  }

  let options = {
    format: 'A4',
    orientation: "landscape",
    border: '1cm',
  };

  htmlPdf.create(html, options).toFile(filename, function(err, res) {
    if (err) {
      fut.return(err);
    } else {
      fut.return("/tmp/Raport.pdf");
    }
  });
  return fut.wait();
}

Meteor.methods({
  foo,
});
```

Client:

```
generateReport() {
    let printArea = angular.element(document.getElementById('report')).html();
    let css = "http://localhost:3000/report.css";
    let html = '<html><html><head><link rel="stylesheet" type="text/css" href="' 
                + css + '"></head><body>' + printArea + '</body></html>'

    Meteor.call('foo', html, this.$bindToContext((error, result) => {
      if (error) {
        alert("Error");
      } else {
        console.log(result);
        //Show "Download" button in your template
      }
    }));
  }
```
